### ISSUE TITLE
Book progress reverts after disconnecting from chromecast

### ISSUE THREAD

**Issue by tboudreaux** [advplyr/audiobookshelf-app#314]
### Steps to reproduce

1. Have some book with non zero progress (either local or streaming)
2. Cast audio to a chrome-cast device and listen for some time 
3. Disconnect ABS from chrome-cast device

### Expected behavior
- Once disconnected from the chrome-cast device the progress of the book in the audio-book shelf app should remain where it was at the time of disconnection

### Actual behavior
- When disconnecting from the chrome-cast device the progress of the book reverts to where it was at the time of initial connection to the chrome-cast. That is, if the book plays back on the chrome-cast starting at 5 hours in and plays for 10 minutes one would expect that when disconnecting the progress of the book would be 5 hours and 10 minutes. However, it is set back to 5 hours.

### Environment data

Audiobookshelf Version: 
Server - Docker 2.1.2
Client - 0.9.55-beta 

- [x] Android App?
- [ ] iOS App?

#### Android Issue
Android version: 12

Device model: Samsung Galaxy S21 Ultra

Stock or customized system: Stock

**Comment by advplyr** [comment]
Confirmed. Starting a cast while the player is open is also broken.

**Comment by advplyr** [comment]
I was able to resolve this for the next release but I also found a deeper issue that will need to be handled eventually.

When casting a local download then disconnecting from the cast it will revert the player in the app to direct playing the media instead of playing it locally.

When switching from cast to device player and vice versa we will need to start a new playback session instead of using the existing one. This will also benefit from distinguishing between playback sessions on chromecast and those on the device.

Currently playing from the cast player then playing from the device player will only show in the server playback session list as the media was played from the device player.
